zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -413,35 +412,47 @@ public static final class AuthorizationRequest {
     private final Set<String> resourceIds;
     private final String tenantId;
     private final boolean isNewResource;
+    private final boolean isTenantOwnedResource;
 
     public AuthorizationRequest(
         final TypedRecord<?> command,
         final AuthorizationResourceType resourceType,
         final PermissionType permissionType,
         final String tenantId,
-        final boolean isNewResource) {
+        final boolean isNewResource,
+        final boolean isTenantOwnedResource) {
       this.command = command;
       this.resourceType = resourceType;
       this.permissionType = permissionType;
       resourceIds = new HashSet<>();
       resourceIds.add(WILDCARD_PERMISSION);
       this.tenantId = tenantId;
       this.isNewResource = isNewResource;
+      this.isTenantOwnedResource = isTenantOwnedResource;
+    }
+
+    public AuthorizationRequest(
+        final TypedRecord<?> command,
+        final AuthorizationResourceType resourceType,
+        final PermissionType permissionType,
+        final String tenantId,
+        final boolean isNewResource) {
+      this(command, resourceType, permissionType, tenantId, isNewResource, true);
     }
```

Comment:
💭 I wonder if we should to more of a builder pattern as opposed to having all these overloaded constructors. But that's a refactoring we can always do at a later point in time.

Reply:
I thought about that as well, but then I would have had to refactor all the processors where the `tenantId` is needed, which would have pushed the scope of this PR a bit.

I created [a task](https://github.com/camunda/camunda/issues/32467) for this.